### PR TITLE
[DOC] Enhanced RDoc for String

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- #start_with?
- #end_with?
- #delete_prefix
- #delete_prefix!
- #delete_suffix
- #delete_suffix!

### CODE REVIEW

File: doc/string/end_with_p.rdoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Returns whether +self+ ends with any of the given +string_or_regexp+.
+
+Matches patterns against the end of +self+.
+For each given +string_or_regexp+, the pattern is:
+
+- +string_or_regexp+ itself, if it is a Regexp.
+- <tt>Regexp.quote(string_or_regexp)</tt>, if +string_or_regexp+ is a string.
```
Comment (peterzhu2118) [inline doc/string/end_with_p.rdoc:7]: I don't think the pattern can be a regex:

```ruby
irb(main):019:0> 'hello'.end_with?(/\d+/)
(irb):19:in `end_with?': no implicit conversion of Regexp into String (TypeError)
```

Interestingly, `String#start_with?` can be a regex:

```ruby
irb(main):021:0> 'hello'.start_with?(/hell/)
=> true
```

File: doc/string/delete_prefix.rdoc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Returns a copy of +self+ with leading substring <tt>prefix</tt> removed:
+
+  'hello'.delete_prefix('hel')      # => "lo"
+  'hello'.delete_prefix('llo')      # => "hello"
+  'тест'.delete_prefix('те')        # => "ст"
+  'こんにちは'.delete_prefix('こん')  # => "にちは"
+
+Related: String#delete_prefix!, String#delete#suffix.
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/string/delete_prefix.rdoc]: Fixed.

File: doc/string/delete_suffix.rdoc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Returns a copy of +self+ with trailing substring <tt>suffix</tt> removed:
+
+  'hello'.delete_suffix('llo')      # => "he"
+  'hello'.delete_suffix('hel')      # => "hello"
+  'тест'.delete_suffix('ст')        # => "те"
+  'こんにちは'.delete_suffix('ちは')  # => "こんに"
+
+Related: String#delete_suffix!, String#delete#prefix.
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/string/delete_suffix.rdoc]: Fixed.

File: doc/string/end_with_p.rdoc (hunk 1 above)
Comment (BurdetteLamar) [reply doc/string/end_with_p.rdoc]: Fixed, here and in call-seq.

**Comment by BurdetteLamar** [comment]
@peterzhu2118, going mostly off the grid for the rest of the week at least -- grandkids duty.

**Pull request merged**